pkg/lib/authn/sso/simple_store.go

```diff
@@ -0,0 +1,67 @@
+package sso
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	goredis "github.com/go-redis/redis/v8"
+
+	"github.com/authgear/authgear-server/pkg/lib/config"
+	"github.com/authgear/authgear-server/pkg/lib/infra/redis/appredis"
+)
+
+type SimpleStoreRedisFactory struct {
+	AppID config.AppID
+	Redis *appredis.Handle
+}
+
+func (f *SimpleStoreRedisFactory) GetStoreByProvider(providerType string, providerAlias string) *SimpleStoreRedis {
+	return &SimpleStoreRedis{
+		redis:         f.Redis,
+		appID:         string(f.AppID),
+		providerType:  providerType,
+		providerAlias: providerAlias,
+	}
+}
+
+func storageKey(appID string, providerType string, providerAlias string, key string) string {
+	return fmt.Sprintf("app:%s:oauth:%s:%s:%s", appID, providerType, providerAlias, key)
+}
+
+type SimpleStoreRedis struct {
+	redis         *appredis.Handle
+	appID         string
+	providerType  string
+	providerAlias string
+}
+
+func (s *SimpleStoreRedis) GetDel(key string) (data string, err error) {
+	ctx := context.Background()
```

Comment:
Inject context.Context in SimpleStoreRedisFactory, and forward it to SimpleStoreRedis. Use that context here, instead of context.Background()